modules/Core/src/main/java/org/terasology/logic/actions/CollisionSystem.java

```diff
@@ -0,0 +1,112 @@
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.CharacterSoundComponent;
+import org.terasology.logic.characters.CharacterSoundSystem;
+import org.terasology.logic.characters.events.HorizontalCollisionEvent;
+import org.terasology.logic.characters.events.VerticalCollisionEvent;
+import org.terasology.logic.health.DoDamageEvent;
+import org.terasology.logic.health.EngineDamageTypes;
+import org.terasology.logic.health.HealthComponent;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class CollisionSystem extends BaseComponentSystem {
+
+    @In
+    private org.terasology.engine.Time time;
+
+    private Random random = new FastRandom();
+
+    /**
+     * React to any vertical collision of an entity that has a
+     * health component and establishes if the character should
+     * take damage and what's the amount
+     * @param event vertical collision
+     * @param entity
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onLand(VerticalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+        float speed = Math.abs(event.getVelocity().y);
+
+        if (speed > health.fallingDamageSpeedThreshold) {
+            int damage = (int) ((speed - health.fallingDamageSpeedThreshold) * health.excessSpeedDamageMultiplier);
+            if (damage > 0) {
+                entity.send(new DoDamageEvent(damage, EngineDamageTypes.PHYSICAL.get(), EntityRef.NULL, EntityRef.NULL));
+            }
+        }
+    }
+
+    /**
+     * Same as method onLand but now it's about horizontal collision
+     * @param event horizontal collision
+     * @param entity
+     */
+    @ReceiveEvent(components = {HealthComponent.class})
+    public void onCrash(HorizontalCollisionEvent event, EntityRef entity) {
+        HealthComponent health = entity.getComponent(HealthComponent.class);
+
+        Vector3f vel = new Vector3f(event.getVelocity());
+        vel.y = 0;
+        float speed = vel.length();
+
+        if (speed > health.horizontalDamageSpeedThreshold) {
+            int damage = (int) ((speed - health.horizontalDamageSpeedThreshold) * health.excessSpeedDamageMultiplier);
+            if (damage > 0) {
+                entity.send(new DoDamageEvent(damage, EngineDamageTypes.PHYSICAL.get(), EntityRef.NULL, EntityRef.NULL));
+            }
+        }
+    }
+
+    /**
+     * For a horizontal collision we check if we can play
+     * the character sounds for landing
+     * @param event vertical collision
```

Comment:
Same with these param tags. For instance `@param event the event that triggered the vertical collision`